### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by FilipChalupa**
I appreciate what you've done. :) But I'm not sure it is working correctly. Can you please check my pull request?

My greatest concern is that previous version wasn't limited to run only once.

### CODE REVIEW

**Comment by mthines** [comment]
Hey @FilipChalupa, and thanks for your PR! 💪 

I totally get your point, and I think perhaps it's more a question of bad naming than actually change of logic :)

I wanted the `useEffect` not to be skipped until the 3rd argument was `true`, and then execute it only once. That means that the logic in the effect will keep on being executed, until it's been executed once with a true 3rd parameter.

If I just want an effect to only execute once, I would use a traditional `useEffect`, which an empty array as dependency.

It looks like with your change that, that won't happen because it defaults to `true`? :)
Could you provide some more information on what you're trying to achieve? :)

**Comment by FilipChalupa** [comment]
I see what you mean. I didn't get that behaviour was intentional. Thank you for explaining that.

I wrongly assumed it should run once and only once. Like run once the condition is met. At least that was what I needed in my project.

Nevertheless your code helped me. :)

I believe you can still benefit from the pull request by reverting 9e02c63.

File: src/index.ts [hunk 1]
```diff
@@ -5,22 +5,20 @@ import { DependencyList, EffectCallback, useEffect, useLayoutEffect, useRef } fr
  * @see https://www.npmjs.com/package/eslint-plugin-react-hooks#advanced-configuration
  */
 
-const depsFallback: DependencyList = [];
```
Comment (mthines) [inline src/index.ts:8]: The reason the dep fallback is out of the hook is to make sure the reference remains the same across rerenders - so it's actually intentionally :)

File: src/index.ts [hunk 2]
```diff
@@ -34,32 +32,32 @@ const depsFallback: DependencyList = [];
...
+  const prevCondition = useRef(false);
 
   useEffect(() => {
     if (prevCondition.current) return;
 
     prevCondition.current = condition ?? true;
 
     return effect();
-  }, [...(deps || depsFallback), condition]);
+  }, [...deps, condition]);
```
Comment (mthines) [inline src/index.ts:44]:
```suggestion
  }, [...(deps || depsFallback), condition]);
```

File: src/index.ts [hunk 3]
```diff
@@ -5,22 +5,20 @@ import { DependencyList, EffectCallback, useEffect, useLayoutEffect, useRef } fr
  * @see https://www.npmjs.com/package/eslint-plugin-react-hooks#advanced-configuration
  */
```
Comment (mthines) [inline src/index.ts:7]:
```suggestion

const depsFallback: DependencyList = [];

```

**Comment by mthines** [comment]
Closing this as I haven't back from you :)

**Pull request closed without merging**